### PR TITLE
support for fetching all JQL results by way of pagination (fixes #1720)

### LINKED ISSUE #1720: JQL: add support for fetching all the results by way of pagination
### Issue Summary

The JQL integration returns only the first 50 issues. This is the default number of issues returned via JIRA REST API. A mechanism should be implemented where a query is executed multiple times to fetch subsequent issues from JIRA.

### Steps to Reproduce

1. Configure JIRA integration
2. Create any JQL query which returns more than 50 issues
3. Execute the query

Expected: More than 50 issues returned.
Actual: Only 50 issues returned.

### Technical details:

* Redash Version: 0.12.0+b2449
* Browser/OS: Ubuntu 16.4
* How did you install Redash: Provisioning script from https://redash.io/help-onpremise/setup/setting-up-redash-instance.html

### PR DESCRIPTION
**Pull request by justmiles**
This PR adds support for fetching all JQL results by way of pagination. I was able to test this against a Jira Cloud instance with successful results.

This resolves #1720

### CODE REVIEW

File: redash/query_runner/jql.py [hunk 1]
```diff
@@ -179,6 +181,19 @@ def run_query(self, query, user):
...
+                while data['total'] > index:
+                    query['startAt'] = index
+                    response, error = self.get_response(jql_url, params=query)
+                    if error is not None:
+                        return None, error
+
+                    data = response.json()
+                    index = data['startAt'] + data['maxResults']
+
+                    addl_results = parse_issues(data, field_mapping)
```
Comment (arikfr) [inline redash/query_runner/jql.py]: Is it possible that the next pages of results might have different columns than the previous ones?

Comment (justmiles) [reply redash/query_runner/jql.py]: Looking at [their docs](https://developer.atlassian.com/server/jira/platform/rest-apis/#pagination) it appears `startAt`, `maxResults`, and `total` will always be in the response.

Comment (justmiles) [reply redash/query_runner/jql.py]: Ah, I see. That wouldn't be possible. They are additional results to the same query. The query defines the columns. If the column is empty it'll still come back but with a null value.

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**